exploration/0477-quoted-literals.md

```diff
@@ -0,0 +1,192 @@
+
+In [#414](https://github.com/unicode-org/message-format-wg/pull/414) proposed to revert these changes and go back to using single and/or double quotes as delimiters. The propsal was rejected. This document is an artifact of that rejection.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+In general, quoted literals are useful for:
+
+1. encoding literals containing whitespace, like literals consisting of multiple words,
+1. encoding literals containing exotic characters that do not conform to the `unquoted` production in ABNF.
+
+More specifically:
+
+- Message authors and translators need to be able to use the apostrophe in the message content, and may want to use the single quote character to represent it instead of the typograhic (curly) apostrophe.
+
+  > ```
+  > {…{|New Year's Eve|}…}
+  > ```
+
+- Message authors may want to use literals to define locale-aware dates as literals in the RFC 7231 format:
+
+  > ```
+  > {The Unix epoch is defined as {|Thu, 01 Jan 1970 00:00:00 GMT| :datetime}.}
+  > ```
+
+- Message authors may want to use multiple words as values of certain options passed to custom functions and markup elements:
+
+  > ```
+  > {{+button title=|Click here!|}Submit{-button}}
+  > ```
+  >
+  > Note that quoted literals cannot contain placeholders, making interpolating data into them impossible.
+  >
+  > ```
+  > -- This is impossible in MessageFormat 2.0.
+  > {{+button title=|Goodbye, {$userName}!|}Sign out{-button}}
+  > ```
+
+- Selector function implementers may want to support exotic characters in variant keys to effectively create "mini-DSLs" for the matching logic:
+
+  > ```
+  > match {$count :myNumber}
+  > when |<10| {A handful.}
+  > when * {Lots.}
+  > ```
+
+- Message authors may want to protect untranslatable strings:
+
+  > ```
+  > {Visit {|http://www.example.com| @translate=false}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Message authors may want to decorate substrings as being written in a particular language, different from the message's language, for the purpose of accessibility, text-to-speech, and semantic correctness.
+
+  > ```
+  > {The official native name of the Republic of Poland is {|Rzeczpospolita Polska| @lang=pl}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Developers may want to embed messages with quoted literals in code written in another programming language which uses single or double quotes to delimit strings.
+
+  > ```js
+  > let message = new MessageFormat("en", "{A message with {|a literal|}.}");
+  > ```
```

Comment:
I think this should be better distinguished from the following JSON example, which necessarily uses double quotes.
```suggestion
- Developers may want to embed messages with quoted literals in code written in another programming language which uses single or double quotes to delimit strings.

  > ```js
  > let message = new MessageFormat('en', '{A message with {|a literal|}.}');
  > ```
```

Reply:
I agree, but unfortunately, it looks like our Prettier bot disagrees: https://github.com/unicode-org/message-format-wg/pull/477/commits/22a4cb2874e1a3ed19747229b72356a55bd9c312. @eemeli, help? :)

Reply:
Filed #481 for this.

Reply:
@eemeli ended up fixing this via a `prettier-ignore` comment, thanks.